.travis.yml

```diff
@@ -55,16 +55,34 @@ install:
   # Customise the testing environment
   # ---------------------------------
   - conda config --add channels conda-forge
-  - if [[ "$TEST_MINIMAL" == true ]]; then
-      conda install --quiet --file minimal-conda-requirements.txt;
+  - >
+    if [[ "$TEST_MINIMAL" == true ]]; then
+      CONDA_REQS_FILE=conda-requirements.txt
+      if [[ "$TRAVIS_PYTHON_VERSION" == 3* ]]; then
+          python requirements/gen_conda_requirements.py --groups test > ${CONDA_REQS_FILE};
+      else
+          python requirements/gen_conda_requirements.py --py2 --groups test > ${CONDA_REQS_FILE};
+      fi
+      cat ${CONDA_REQS_FILE};
+      conda install --quiet --file ${CONDA_REQS_FILE};
     else
+      CONDA_REQS_FILE=conda-requirements.txt
       if [[ "$TRAVIS_PYTHON_VERSION" == 3* ]]; then
-        sed -e '/esmpy/d' -e 's/#.\+$//' conda-requirements.txt | xargs conda install --quiet;
+          python requirements/gen_conda_requirements.py --groups test full > ${CONDA_REQS_FILE};
       else
-        conda install --quiet --file conda-requirements.txt;
+          python requirements/gen_conda_requirements.py --py2 --groups test full > ${CONDA_REQS_FILE};
       fi
+      cat ${CONDA_REQS_FILE};
+      conda install --quiet --file ${CONDA_REQS_FILE};
     fi
 
+
+  - >
+    if [[ "${TEST_TARGET}" == 'doctest' ]]; then
+        conda install --quiet --file requirements/docs.txt
```

Comment:
@pelson `requirements/docs.txt` doesn't seem to exist ...

Reply:
Done.